src/Control/Carrier/Readline/Haskeline.hs

```diff
@@ -25,10 +26,18 @@ import System.Environment
 import System.FilePath
 import System.IO (stdout)
 
+#if MIN_VERSION_haskeline(0, 8, 0)
 runReadline :: (MonadIO m, MonadMask m) => Prefs -> Settings m -> ReadlineC m a -> m a
+#else
+runReadline :: (MonadIO m, MonadMask m, MonadException m) => Prefs -> Settings m -> ReadlineC m a -> m a
```

Comment:
I think the `MonadMask` constraint (and the dependency on `exceptions`, for that matter) is _actually_ redundant on 0.7.